internal/rbd/rbd_util.go

```diff
@@ -651,74 +624,77 @@ func (rv *rbdVolume) ensureImageCleanup(ctx context.Context) error {
 		if val.Name == rv.RbdImageName {
 			rv.ImageID = val.Id
 
-			return trashRemoveImage(ctx, rv, rv.conn.Creds)
+			return rv.trashRemoveImage(ctx)
 		}
 	}
 
 	return nil
 }
 
 // deleteImage deletes a ceph image with provision and volume options.
-func deleteImage(ctx context.Context, pOpts *rbdVolume, cr *util.Credentials) error {
-	image := pOpts.RbdImageName
+func (rv *rbdVolume) deleteImage(ctx context.Context) error {
```

Comment:
make this a method of `rbdImage` struct?

Reply:
that is reported as #2742